pkpdapp/pkpdapp/models/subject.py

```diff
@@ -0,0 +1,27 @@
+#
+# This file is part of PKPDApp (https://github.com/pkpdapp-team/pkpdapp) which
+# is released under the BSD 3-clause license. See accompanying LICENSE.md for
+# copyright notice and full license details.
+#
+
+from django.db import models
+from jsonfield import JSONField
+from pkpdapp.models import Dataset
+
+
+class Subject(models.Model):
+    """
+    A subject in a a particular dataset.
+    """
+    id_in_dataset = models.IntegerField(help_text='unique id in the dataset')
+    dataset = models.ForeignKey(
+        Dataset, on_delete=models.CASCADE,
+        help_text='dataset containing this subject'
+    )
+    metadata = JSONField(help_text='subject metadata')
+
+    class Meta:
+        constraints = [
+            models.UniqueConstraint(fields=['id_in_dataset', 'dataset'],
```

Comment:
I'm not sure I understand this: is this to check that the dataset contains a subject id 'id_in_dataset'?